### PR TITLE
Tag name filtering

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
Adds a `q=` parameter to the `/api/v2/search/tags` endpoint that allows for filtering down tag names in the same way as the tag values endpoint.

Notes:
- Instead of pulling and parsing all dedicated and well known columns we are just including them if they have any values.

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/api_docs/_index.md [hunk 1]
```diff
@@ -326,6 +326,8 @@ Parameters:
 - `scope = (resource|span|intrinsic)`
   Specifies the scope of the tags, this is an optional parameter, if not specified it means all scopes.
   Default = `all`
+- `q = (traceql query)`
+  Optional. A TraceQL query to filter tag names by. Currently only works for a single spanset of &&ed conditions. ie. `{ span.foo = "bar" && resource.baz = "bat" ...}`. See also [Filtered tag values](#filtered-tag-values).
```
Comment (knylander-grafana) [inline docs/sources/tempo/api_docs/_index.md:330]:
```suggestion
  Optional. A TraceQL query to filter tag names by. Currently only works for a single spanset of `&&`ed conditions. For example: `{ span.foo = "bar" && resource.baz = "bat" ...}`. See also [Filtered tag values](#filtered-tag-values).
```

**Review by knylander-grafana — COMMENTED** [review]
Thank you for updating the docs! I only had a minor suggestion.

File: tempodb/encoding/vparquet3/block_autocomplete_test.go [hunk 2]
```diff
@@ -324,6 +550,29 @@ func BenchmarkFetchTagValues(b *testing.B) {
 			tag:   "resource.namespace",
 			query: `{span.http.status_code=200}`,
 		},
+		{
```
Comment (ie-pham) [inline tempodb/encoding/vparquet3/block_autocomplete_test.go:553]: duplicated from line 550?

File: tempodb/encoding/vparquet4/block_autocomplete_test.go [hunk 3]
```diff
@@ -324,6 +550,29 @@ func BenchmarkFetchTagValues(b *testing.B) {
 			tag:   "resource.namespace",
 			query: `{span.http.status_code=200}`,
 		},
+		{
```
Comment (ie-pham) [inline tempodb/encoding/vparquet4/block_autocomplete_test.go:553]: dupe from 550?

**Review by ie-pham — APPROVED** [review]
Looks good! just very small comments. Send it

**Comment by ie-pham** [comment]
Remember to let the front end folks know about the new feature ;)

File: tempodb/encoding/vparquet3/block_autocomplete_test.go (hunk 2 above)
Comment (joe-elliott) [reply tempodb/encoding/vparquet3/block_autocomplete_test.go:553]: removed.

**Pull request merged**